students/guillaume/quarter_2_project/leankitbackend/leankitbackend/leankitcards.py

```diff
@@ -0,0 +1,297 @@
+
+'''
+Maybe define a Project class
+'''
+
+lanes_current_sprint = ['Current Sprint Backlog', 'Doing Now', 'Under Review',
+                        'Finished As Planned']
+
+lanes_Backlogs_next_sprint = ['Backlog', 'Next Sprint Planning']
+
+
+# Attention sub columns of recently finished / ready to archived has the
+# same name
+lanes_Finished = ['Finished As Planned', 'Started but not Finished',
+                  'Discarded or Obsolete']
+
+
+class Leankit_Cards:
+    def __init__(self, cards_list=None, date=None):
+        self.date = date
+        if cards_list is not None:
+            self.append(cards_list, date)
+
+    def append(self, cards_list, date=None):
+        if isinstance(cards_list, list):
+            self.date = date
+            self.cards_list = []
+            self.cards_id = []
+            self.cards_title = []
+            self.json = []
+
+            try:
+                for card in cards_list:
+                    self.cards_list.append(Leankit_Card(card))
+                    self.cards_id.append(card['id'])
+                    self.cards_title.append(card['title'])
+                    self.json.append(self.cards_list[-1].json)
+
+                self.projects = projects(self.cards_list)
+                self.len = len(cards_list)
+                self.status = True
+                # self.json = [card.json for card in self.cards_list]
+            except KeyError:
+                self.status = False
+
+    def projects(self):
+        return projects(self)
+
+    def sorted_projects(self):
+        return sorted_projects(self)
+
+    def in_progress(self):
+        return self.sorted_projects()['In Progress']
+
+    @property
+    def date(self):
+        return self._date
+
+    @date.setter
+    def date(self, datetime_obj):
+        if isinstance(datetime_obj, datetime):
+            self._date = datetime_obj
+            # self.year = self.date.year
+            # self.week_n = self.date.isocalendar[1]
+
+    def __repr__(self):
+        if hasattr(self, 'cards_list'):
+            return ('').join([card.__repr__() for card in self.cards_list])
+        return 'There is no cards being imported'
+
+
+class Leankit_Card:
+    def __init__(self, card):
+        if isinstance(card, dict):
+            self.json = card
+            for key, value in card.items():
+                setattr(self, key, value)
+            self.status = True
+        else:
+            self.status = False
+
+    def __repr__(self):
+        try:
+            Name = ''
+            if len(self.assignedUsers) > 0:
+                tmp_lst = [user['fullName'] for user in self.assignedUsers]
+                Name = (', ').join(tmp_lst)
+            tmp_lst = self.type['title'], self.id, self.title, Name
+            return '{} {} {} {}\n'.format(*tmp_lst)
+        except (AttributeError, KeyError):
+            return 'input was not a card'
+
+
+class Leankit_Board:
+    def __init__(self, board):
+        if isinstance(board, dict):
```

Comment:
you probably don't need to check for a type at all -- but if you do, you want to use:

`if instance(collections.abc.Mapping): `

We didn't get into Abstract Base Classes in class, as they aren't very widely used, but it's a way to check that you've got a "mapping" type, which would be a dict, but also any other type that acts like a dict.